crypto/txsigner_homestead.go

```diff
@@ -0,0 +1,103 @@
+package crypto
+
+import (
+	"crypto/ecdsa"
+	"errors"
+	"math/big"
+
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+// HomesteadSigner may be used for signing pre-EIP155 transactions
+type HomesteadSigner struct {
+	FrontierSigner
+}
+
+// NewHomesteadSigner returns new FrontierSigner object (constructor)
+//
+// HomesteadSigner accepts the following types of transactions:
+//   - pre-EIP-155 transactions
+func NewHomesteadSigner() *HomesteadSigner {
+	return &HomesteadSigner{
+		FrontierSigner: FrontierSigner{},
+	}
+}
+
+// Hash returns the keccak256 hash of the transaction
+//
+// The pre-EIP-155 transaction hash preimage is as follows:
+// RLP(nonce, gasPrice, gas, to, value, input)
+//
+// Specification: https://eips.ethereum.org/EIPS/eip-155#specification
+//
+// Note: Since the hash is calculated in the same way as with FrontierSigner, this is just a wrapper method
+func (signer *HomesteadSigner) Hash(tx *types.Transaction) types.Hash {
+	return signer.FrontierSigner.Hash(tx)
+}
+
+// Sender returns the sender of the transaction
+func (signer *HomesteadSigner) Sender(tx *types.Transaction) (types.Address, error) {
+	if tx.Type() != types.LegacyTx {
+		return types.Address{}, errors.New("Sender method: Unknown transaction type")
```

Comment:
Add a new error constant:

`	ErrTxTypeNotSupported   = errors.New("transaction type not supported")`

And use it in places where you receive an unexpected tx type.